search/search-client-opensearch/src/main/java/io/camunda/search/transformers/query/MatchQueryTransformer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.transformers.query;
+
+import io.camunda.search.clients.query.SearchMatchQuery;
+import io.camunda.search.transformers.OpensearchTransformers;
+import org.opensearch.client.opensearch._types.FieldValue;
+import org.opensearch.client.opensearch._types.query_dsl.MatchQuery;
+import org.opensearch.client.opensearch._types.query_dsl.Operator;
+import org.opensearch.client.opensearch._types.query_dsl.QueryBuilders;
+
+public final class MatchQueryTransformer
+    extends QueryOptionTransformer<SearchMatchQuery, MatchQuery> {
+
+  public MatchQueryTransformer(final OpensearchTransformers transformers) {
+    super(transformers);
+  }
+
+  @Override
+  public MatchQuery apply(final SearchMatchQuery value) {
+    final var field = value.field();
+    final var query = value.query();
+    final var queryAsFieldValue = FieldValue.of(query);
+    final var operator = Operator.valueOf(value.operator());
```

Comment:
Should we have input validation here?

```
Objects.requireNonNull(value.field(), "field must not be null");
Objects.requireNonNull(value.query(), "query must not be null");
Objects.requireNonNull(value.operator(), "operator must not be null");
```

I am not 100% sure it is applicable on this case, but just in order to avoid Null Pointer Exceptions. WDYT?

Reply:
Only the `query` is required; this is already checked in the search-client itself when building the `SearchMatchQuery`. The `operator` can be null. Regarding the `operator`, I introduced an enum that way only allowed values can be set. That way, it should away any NPEs.